aries/aries_vcx_core/src/wallet/askar/packing.rs

```diff
@@ -0,0 +1,303 @@
+use aries_askar::{
+    crypto::alg::Chacha20Types,
+    kms::{KeyAlg, KeyAlg::Ed25519, KeyEntry, LocalKey, ToDecrypt},
+    Session,
+};
+use public_key::{Key, KeyType};
+
+use super::{
+    askar_utils::{
+        bs58_to_bytes, bytes_to_bs58, bytes_to_string, ed25519_to_x25519_pair,
+        ed25519_to_x25519_private, ed25519_to_x25519_public, from_json_str,
+        local_key_to_private_key_bytes, local_key_to_public_key_bytes,
+    },
+    packing_types::{
+        AnoncryptRecipient, AuthcryptRecipient, Base64String, Jwe, JweAlg, ProtectedData,
+        ProtectedHeaderEnc, ProtectedHeaderTyp, Recipient,
+    },
+};
+use crate::{
+    errors::error::{AriesVcxCoreError, AriesVcxCoreErrorKind, VcxCoreResult},
+    wallet::{
+        askar::crypto_box::{CryptoBox, SodiumCryptoBox},
+        structs_io::UnpackMessageOutput,
+    },
+};
+
+pub struct Packing {
+    crypto_box: Box<dyn CryptoBox + Send + Sync>,
+}
+
+impl Packing {
+    pub fn new() -> Self {
+        Self {
+            crypto_box: Box::new(SodiumCryptoBox::new()),
+        }
+    }
+
+    pub async fn unpack(
+        &self,
+        jwe: Jwe,
+        session: &mut Session,
+    ) -> VcxCoreResult<UnpackMessageOutput> {
+        let protected_data = self.unpack_protected_data(&jwe)?;
+        let (recipient, key_entry) = self.find_recipient_key(&protected_data, session).await?;
+        let local_key = key_entry.load_local_key()?;
+        let (enc_key, sender_verkey) = self.unpack_recipient(recipient, &local_key)?;
+        Ok(UnpackMessageOutput {
+            message: self.unpack_msg(&jwe, enc_key)?,
+            recipient_verkey: recipient.unwrap_kid().to_owned(),
+            sender_verkey: sender_verkey.map(|key| key.base58()),
+        })
+    }
+
+    pub fn unpack_protected_data(&self, jwe: &Jwe) -> VcxCoreResult<ProtectedData> {
```

Comment:
Like other functions in `Packing`, this is also taking, but not actually using `&self`